src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -623,7 +623,8 @@ private BoundExpression BindSimpleBinaryOperator(BinaryExpressionSyntax node, Di
                     HashSet<DiagnosticInfo> useSiteDiagnostics = null;
                     bool leftDefault = left.IsLiteralDefault();
                     bool rightDefault = right.IsLiteralDefault();
-                    foundOperator = !isObjectEquality || BuiltInOperators.IsValidObjectEquality(Conversions, leftType, leftNull || leftDefault, rightType, rightNull || rightDefault, ref useSiteDiagnostics);
+                    foundOperator = !isObjectEquality || BuiltInOperators.IsValidObjectEquality(Conversions,
```

Comment:
>IsValidObjectEquality [](start = 74, length = 21)

This probably needs to be revised in light of latest LDM decision (ie. allow usage in binary operators).